src/kirby/kirby.common.ts

```diff
@@ -50,8 +54,15 @@ export const declarations = [
   AvatarComponent,
   GroupByPipe,
   SpinnerComponent,
+  ModalComponent,
   SegmentedControlComponent,
   ChipComponent,
   SegmentedChipControlComponent,
   BadgeComponent,
 ];
+
+export const providerDeclarations: any[] = [
+  ModalController,
+  { provide: IModalController, useExisting: ModalController },
```

Comment:
Consider adding a comment, that this declaration fixes cyclic reference problem...